Apps/WebClient/src/ClientApp/src/views/timeline.vue

```diff
@@ -489,6 +502,54 @@ export default class TimelineView extends Vue {
             });
     }
 
+    private fetchComments() {
+        this.isCommentLoading = true;
+
+        this.retrieveProfileComments({
+            hdid: this.user.hdid,
+        })
+            .then((results) => {
+                if (results.resultStatus == ResultType.Success) {
+                    this.logger.debug("Profile Comments Loaded");
+                } else {
+                    this.logger.error(
+                        "Error returned from the retrieve comments call: " +
+                            JSON.stringify(results.resultError)
+                    );
+                    this.addError(
+                        ErrorTranslator.toBannerError(
+                            "Profile Comments Error",
+                            results.resultError
+                        )
+                    );
+                }
+            })
+            .catch((err) => {
+                this.logger.error(err);
+                this.addError(
+                    ErrorTranslator.toBannerError("Profile Comments Error", err)
+                );
+            })
+            .finally(() => {
+                this.isCommentLoading = false;
+            });
+    }
+
+    private loadImmunizationEntries() {
+        if (this.immunizationLoadReady) {
+            this.immunizationLoadDeferred = false;
+            this.immunizationLoadReady = false;
+        }
+        // Add the immunization entries to the timeline list
+        for (let immunization of this.patientImmunizations) {
```

Comment:
We should switch to this pattern to all other data as well. Request from Vuex for the load and have the vue reactivity update the timelineEntries array as it gets updated. That way the logic for populating is centralized and vuex is reponsable of maintaining state of the timeline